Firestore/core/src/firebase/firestore/core/relation_filter.cc

```diff
@@ -47,22 +49,22 @@ bool RelationFilter::Matches(const model::Document& doc) const {
 
 bool RelationFilter::MatchesValue(const FieldValue& other) const {
   // Only compare types with matching backend order (such as double and int).
-  return FieldValue::Comparable(value_.type(), other.type()) &&
+  return FieldValue::Comparable(value_rhs_.type(), other.type()) &&
```

Comment:
This now looks wrong (it's `value_rhs_` but it's on the left).

Reply:
fixed.